tests/functional/modules/test_zos_volume_init_func.py

```diff
@@ -16,37 +16,46 @@
 
 __metaclass__ = type
 
-# TEST_VOL_ADDR = '0903'
-# TEST_VOL_SER = 'KET999'
-TEST_VOL_ADDR = '01A2'
-TEST_VOL_SER = 'USER02'
+from ibm_zos_core.tests.helpers.volumes import Volume_Handler
+from ibm_zos_core.tests.helpers.dataset import get_tmp_ds_name
+
+# To run the test suite is a recommendation to use the volumes in order
+# 'USER02', '01A2'
+# new list of similar volumes will be added in next iteration
 
 INDEX_CREATION_SUCCESS_MSG = 'VTOC INDEX CREATION SUCCESSFUL'
 VTOC_LOC_MSG = "ICK01314I VTOC IS LOCATED AT CCHH=X'0000 0001' AND IS  {:4d} TRACKS."
 
+def clear_volume(hosts, volume):
+    datasets_in_volume = hosts.all.shell(cmd="vtocls {0}".format(volume))
+    for dataset in datasets_in_volume.contacted.values():
+        datasets = str(dataset.get("stdout")).split("\n")
+    for dataset in datasets:
+        dataset_t_del = dataset.split(' ', 1)[0]
+        hosts.all.shell(cmd="""drm "{0}" """.format(dataset_t_del))
 
 # Guard Rail to prevent unintentional initialization of targeted volume.
 # If this test fails, either reset target volume serial to match
 # verify_volid below or change value to match current volume serial on
 # target.
 
-def test_guard_rail_and_setup(ansible_zos_module):
+def test_guard_rail_and_setup(ansible_zos_module, volumes_on_systems):
     hosts = ansible_zos_module
-
     # remove all data sets from target volume. Expected to be the following 3
-    hosts.all.zos_data_set(name="IMSTESTL.IMS01.SPOOL1", state="absent")
-    hosts.all.zos_data_set(name="IMSTESTL.IMS01.SPOOL2", state="absent")
-    hosts.all.zos_data_set(name="IMSTESTL.IMS01.SPOOL3", state="absent")
+    volumes = Volume_Handler(volumes_on_systems)
+    volume_1 = volumes.get_available_vol()
+    volume_2 = volumes.get_available_vol()
+    clear_volume(hosts, volume_1)
```

Comment:
do you need to check that volume_1 is not the same as volume_2?

Reply:
At line 22 I already add comment that even if is not hardcoded, these two volumes that we use are still need to be added in our test_config file, because are so specific the characteristics of the volume

Reply:
I didn't notice the comment.  My bad.